ludwig/hyperopt/execution.py

```diff
@@ -585,6 +588,103 @@ def execute(
         return hyperopt_results
 
 
+class RayTuneExecutor(HyperoptExecutor):
+    def __init__(self, hyperopt_sampler, output_feature, metric, split, goal, **kwargs):
+        ray.init(ignore_reinit_error=True)
+        HyperoptExecutor.__init__(self, hyperopt_sampler, output_feature,
+                                  metric, split)
+        self.search_space = hyperopt_sampler
+        self.output_feature = output_feature
+        self.metric = metric
+        self.split = split
+        self.goal = goal
+        self.trial_id = 0
+        self.experiment_kwargs = {}
+
+    def _run_experiment(self, config):
+
+        self.trial_id += 1
```

Comment:
This won't work since the function passed into `tune.run` will be duplicated and executed on multiple processes (therefore, `self` will not reference the same object)

Reply:
Ahh, @richardliaw is there a way to enumerate the processes and have it accessible inside the function?

Reply:
Hmm, it might be difficult. Just curious, what are you trying to achieve by doing this?

Reply:
The model and statistics are saved in a directory `[output_dir]/[experiment_name]_[model_name]_trial_id` where all variables are resolved to user-specified ones and `trial_id` is an increasing number starting from 0 used to differentiate among repeated runs.

Reply:
Ah, I see. Does `trial_id` need to be numerical? In Tune, we have a unique identifier for trial_id but it is a hex value.

Reply:
Oh, that's the convention we have been following. @w4nderlust Would like your take on this?

Reply:
The hex may work fine. It's basically just a way to avoid saving to the same directory when running on the same filesystem or on a shared filesystem. Unique hexes would achieve the same goal ;)

Reply:
Nice!